### DISCUSSION TITLE
UnitTesting with Spectre.Console.Cli

### DISCUSSION THREAD

**Discussion by ghost** [spectreconsole/spectre.console#1398] [Q&A] [answered]
Hi there! Is there any way to call the Execute method without the apps.Run(args);? And if yes. How can I put the right parameters in the constructors of the CommandContext and CommandSettings class?

**Comment by patriksvensson** [comment, marked as answer]
I would recommend using the Spectre.Console.Testing library, and use that to test your commands.

Most of Spectre.Console's own tests uses that library.

**Reply by ghost** [reply]
Ok, Thanks
